sdcm/utils/gce_utils.py

```diff
@@ -271,3 +261,275 @@ def _get_log_entries(self, service, query, page_token=None):
+) -> Any:
+    """
+    Waits for the extended (long-running) operation to complete.
+
+    If the operation is successful, it will return its result.
+    If the operation ends with an error, an exception will be raised.
+    If there were any warnings during the execution of the operation
+    they will be printed to sys.stderr.
+
+    Args:
+        operation: a long-running operation you want to wait on.
+        verbose_name: (optional) a more verbose name of the operation,
+            used only during error and warning reporting.
+        timeout: how long (in seconds) to wait for operation to finish.
+            If None, wait indefinitely.
+
+    Returns:
+        Whatever the operation.result() returns.
+
+    Raises:
+        This method will raise the exception received from `operation.exception()`
+        or RuntimeError if there is no exception set, but there is an `error_code`
+        set for the `operation`.
+
+        In case of an operation taking longer than `timeout` seconds to complete,
+        a `concurrent.futures.TimeoutError` will be raised.
+    """
+    result = operation.result(timeout=timeout)
+
+    if operation.error_code:
+        LOGGER.debug("Error during %s: [Code: %s]: %s", verbose_name,
+                     operation.error_code, operation.error_message)
+        LOGGER.debug("Operation ID: %s", operation.name)
+        raise operation.exception() or RuntimeError(operation.error_message)
+
+    if operation.warnings:
+        LOGGER.debug("Warnings during %s:", verbose_name)
+        for warning in operation.warnings:
+            LOGGER.debug(" - %s: %s", warning.code, warning.message)
+
+    return result
+
+
+def disk_from_image(  # pylint: disable=too-many-arguments
+    disk_type: str,
+    boot: bool,
+    disk_size_gb: int = None,
+    source_image: str = None,
+    auto_delete: bool = True,
+    device_name: str = None,
+    type_: str = None,
+    interface: str = None,
+) -> compute_v1.AttachedDisk:
+    """
+    Create an AttachedDisk object to be used in VM instance creation. Uses an image as the
+    source for the new disk.
+
+    Args:
+         disk_type: the type of disk you want to create. This value uses the following format:
+            "zones/{zone}/diskTypes/(pd-standard|pd-ssd|pd-balanced|pd-extreme)".
+            For example: "zones/us-west3-b/diskTypes/pd-ssd"
+        disk_size_gb: size of the new disk in gigabytes
+        boot: boolean flag indicating whether this disk should be used as a boot disk of an instance
+        source_image: source image to use when creating this disk. You must have read access to this disk. This can be one
+            of the publicly available images or an image from one of your projects.
+            This value uses the following format: "projects/{project_name}/global/images/{image_name}"
+        auto_delete: boolean flag indicating whether this disk should be deleted with the VM that uses it
+
```

Comment:
The most unclear `type_` parameter is not described here.

Reply:
added docstring for this parameter as well